### ISSUE TITLE
Please cater versionCode on updates

### ISSUE THREAD

**Issue by IzzySoft** [nerone-github/WebViewAuto#17]
Could you please increase `versionCode` as well if you present a newer APK? Otherwise Android doesn't recognize it as an update (same for your other 2 Auto apps).

Thanks in advance!

**Comment by nerone-github** [comment]
Hm, basically i would need to change this and sync the gradle each time. Recently i added a versioning based on the date so it should be always possible to see if the apk is newer or not (Installing on top is always possible), so if you check the version (e.g. V 18.01.19-2159), and the last apk was commited on January,20 you know that you are missing an update. And thats even possible without downloading the apk in the first place. So i'm not so sure that it would be a big help (other than taking more time to build ;))

**Comment by IzzySoft** [comment]
Oh, it would. But maybe I should give a little [background on why](https://apt.izzysoft.de/fdroid/index/apk/org.openauto.webviewauto): as long as `versionCode` is the same, fdroidserver considers two `.apk` files duplicates – so the newer will be ignored, as "we already have it". Overriding this manually (by removing the older file) has not much effect: the newer file would make it into the repo alright, but users wouldn't get notified there's an update (as "it isn't, according to `versionCode`").

For "manual updates" it might be clear due to the file name – but I'm speaking about automated updates here :wink:

**Comment by nerone-github** [comment]
But i'm not using Fdroid to manage my apk's and to be honest i don't really like that mirror you created with even offering the apk to download on a third party site (especially on *.de domain). I'd like to keep it on Github completely. Especially i don't like that fact that you did not even ask. But well thats the drawback of github ;) i can't forbid it. I don't think it is a good idea to offer it on a mirror, since it is built with an unofficial sdk, i don't want it to have it wide-spread public attention. Especially you should be careful offering it on a *.de domain.

**Comment by IzzySoft** [comment]
Sorry for not having asked beforehand – usually it's either warmly welcome, or I never receive any feedback; and being open source means it's open to everyone. But of course I honor the authors' requests: if you don't want me to include your apps with my repo, I'll remove them immediately. Intention of my repo is simply to make it easier for users to a) find apps, b) install them, c) keep them up-to-date without having to rely on Playstore or scan Github.

**TL;DR:** If you want me to remove your apps from my repo (I guess all 3 of them then), just let me know and I'll comply. If at a later point you want me to pick them up again, drop me a note: I'll do my best to remember to ***not*** add them back between those times – and apply that to any additional app of yours.

**Comment by nerone-github** [comment]
I'd rather check if i can add them on F-droid repository (However i've read that it is not allowed to post apps which use com.google classes). 

Not that I'm not saying this to harass you or users, but i simply don't want to take any liability risk, and especially posting them on a *.de domain can cause issues. I'm not a lawyer but i'm aware that anything posted on a german site might cause the content to be subject of german federal law. And i see 2 problematic things here - 1.) The app is build with an unofficial APK. 2.) It  might cause car crashes due to neglectant users, and if anything happens which is related to the App, they might ask - from where is it from? It's from IzzySoft! -> Ahh.. ok we contact him about liability.. Instead stuff posted on Github.com is more anonyme. You know what i mean? So I'm also saying this to "protect" you or your company if it is a company site. Unless you can be sure that from a legal point of view everything should be fine, I would advise, that you remove it. And as for finding a way to update the apk's we can think of a better solution. Maybe I can implement an integrated "Update apk" service from within the app which checks the github repo apk. How does that sound?

**Comment by IzzySoft** [comment]
> I'd rather check if i can add them on F-droid repository (However i've read that it is not allowed to post apps which use com.google classes).

I see no show-stopper in the libraries used. Not all `com.google.*` classes are counted as "proprietary" (AFAIK, e.g. Google GSon is considered "OK") – things like GA, GMS and Firebase would be rejected. My repo is considered a "step stone" to the official one for projects "not yet ready, but on their way" even by the maintainers of F-Droid (and once they've made it there, they are usually removed from my repo to avoid confusion caused by different signatures – unless "reproducible builds" were used where that confusion doesn't come up; in those cases I mostly leave the decision to the authors, as my repo usually has the updates much earlier (I don't build myself, so I have no build server issues)).

> Note that I'm not saying this to harass you or users …

I didn't read it that way, no worries. The way you've phrased it made it clear to me you had other reasons – though I didn't know exactly which, it sounded like you not only want to protect yourself but me as well ("especially on *.de domain" let me assume such). And I can see your reasoning: 1) some services don't like being addressed by 3rd party clients (I remember Moxie's reaction on LibreSignal, for example) – and for 2) I made sure to include your UPPER CASE warnings :wink: Just because an app is called something with "car" doesn't mean the dev (or "hoster") has to take responsibility for how users interact with it while driving a car. Same warnings must apply to ALL apps one uses while driving a car, be it media players or messaging apps. Otherwise you could blame car manufacturers if on a bank robbery a car of their brand was used by the robbers to escape. But agreed, "common sense" and "laws" do not always go hand-in-hand :scream:

So yes, I know what you mean! And the site is private: I'm a one-man-company; though [www.izzysoft.de](https://www.izzysoft.de/) is used for my "business stuff", other *.izzysoft.de sub-domains are used for "hobby stuff", like [android.izzysoft.de](https://android.izzysoft.de/) for Android stuff (I've authored some books on Android targeted at users, and also am teaching classes for them) where the repo "belongs to".

Sure, an integrated update service would solve the auto-updates (checking maybe once a day should suffice, plus an option to turn it off as soon as it is served by some "official repo") – which just leaves the "finding-the-app" part open. But agreed: maybe I'd rather not risk the "legal risk", unlogically as it sounds in terms of "common sense". Especially as I myself don't even own a car, so I've no "personal interest" anyway – and it's not always worth risking punishment just for helping others :wink:
